src/Combinatorics/Graphs/functions.jl

```diff
@@ -169,9 +331,20 @@ function add_vertex!(g::Graph{T}) where {T <: Union{Directed, Undirected}}
     return n_vertices(g) - 1 == old_nvertices
 end
 
+function add_vertex!(mg::MixedGraph)
+  @assert add_vertex!(_directed_component(mg))
```

Comment:
```suggestion
  add_vertex!(_directed_component(mg)) || return false
```
this may already fail, but in that case it is ok to just return false and we don't need to throw an error